addons/mail/static/src/core/common/message_model.js

```diff
@@ -255,7 +255,7 @@ export class Message extends Record {
     }
 
     get resUrl() {
-        return `${url("/web")}#model=${this.thread.model}&id=${this.thread.id}`;
+        return `${url("/odoo")}/${this.thread.model}/${this.thread.id}`;
```

Comment:
Can't we have a helper (takin model and id) to build these URLs? It seems we have to manually do it in many places.

Reply:
You can use `stateToUrl` in JS if you don't want to create it by hand.

```js
expect(stateToUrl({ model: "some.model", resId: 2 })).toBe("/odoo/some.model/2");
```